packages/admin-ui/src/combobox/combobox.state.ts

```diff
@@ -57,44 +56,50 @@ export function useComboboxState<T>(
     }
   }, [state.value])
 
-  const status = useMemo<Status>(() => {
-    if (loading) {
-      return 'loading'
-    }
-
-    if (error) {
-      return 'error'
+  const status = useMemo<ComboboxStatus>(() => {
+    if (userStatus) {
+      return userStatus
     }
 
     const noMatches = !matches.length
 
     if (noMatches && state.value === '') {
-      return 'empty-search'
+      return 'empty'
     }
 
     if (noMatches) {
-      return 'no-result'
+      return 'not-found'
     }
 
     return 'ready'
-  }, [loading, error, matches, state.value])
+  }, [userStatus, matches, state.value])
 
   return {
     ...state,
     deferredValue,
-    setError,
-    setLoading,
     status,
+    setStatus,
     getOptionValue,
     renderOption,
     setSelectedItem,
     selectedItem,
     setMatches,
     matches,
+    setError: (bool) => {
+      setStatus(bool ? 'error' : undefined)
+    },
+    setLoading: (bool) => {
+      setStatus(bool ? 'loading' : undefined)
+    },
```

Comment:
This API is unnecessary. `setStatus` is enough to handle this

Reply:
I thought about keeping it for a while just so I don't break the current docs for people reading about combobox